packages/cli/src/models/network/NetworkController.ts

```diff
@@ -239,8 +249,15 @@ export default class NetworkController {
     const libName = libClass.schema.contractName;
     await this._setSolidityLibs(libClass); // Libraries may depend on other libraries themselves
     Loggy.spin(__filename, '_uploadSolidityLib', `upload-solidity-lib${libName}`, `Uploading ${libName} library`);
-    const libInstance = await this.project.setImplementation(libClass, libName);
-    this.networkFile.addSolidityLib(libName, libInstance);
+
+    if (this.project !== undefined) {
+      const libInstance = await this.project.setImplementation(libClass, libName);
+      this.networkFile.addSolidityLib(libName, libInstance);
+    } else {
+      const libInstance = await Transactions.deployContract(libClass);
```

Comment:
Can you add a comment explaining _when_ there is no project? Outside the context of this PR it's not clear.